src/mongo/db/write_concern_options_test.cpp

```diff
@@ -123,76 +120,62 @@ TEST(WriteConcernOptionsTest, ParseSetsWNumNodesIfWIsANumber) {
     auto sw = WriteConcernOptions::parse(BSON("w" << 3));
     ASSERT_OK(sw.getStatus());
     WriteConcernOptions options = sw.getValue();
-    ASSERT_TRUE(WriteConcernOptions::SyncMode::UNSET == options.syncMode);
-    ASSERT_EQUALS(3, options.wNumNodes);
-    ASSERT_EQUALS("", options.wMode);
-    ASSERT_EQUALS(0, options.wTimeout);
-}
-
-TEST(WriteConcernOptionsTest, ParseSetsWTimeoutToZeroIfWTimeoutIsNotANumber) {
```

Comment:
This represents a behavioral change: we used to coerce a string value for `wTimeout` into 0, and now we will report an error. It's not very difficult to support the old behavior, but upon reflection it seems like this would always have been user error in practice so I wanted to leave the door open for fixing that if possible.